base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -264,12 +264,11 @@ where TBackend: OutputManagerBackend + 'static
                 self.add_output(uo).await.map(|_| OutputManagerResponse::OutputAdded)
             },
             OutputManagerRequest::GetBalance => {
-                let tip = self
-                    .base_node_service
-                    .get_connected_base_node_state()
-                    .await?
-                    .height_of_longest_chain();
-                self.get_balance(Some(tip)).await.map(OutputManagerResponse::Balance)
+                let tip_or_none = match self.base_node_service.get_connected_base_node_state().await {
+                    Ok(metadata) => Some(metadata.height_of_longest_chain()),
+                    Err(_) => None,
+                };
+                self.get_balance(tip_or_none).await.map(OutputManagerResponse::Balance)
```

Comment:
A neater Rust way to do this:
```Rust
self.get_balance(self.base_node_service.get_connected_base_node_state().await.ok()).await.map(OutputManagerResponse::Balance)
```

Reply:
Yeah I was just wondering about `.ok()` - but you still need to call `height_of_longest_chain()` ?

Reply:
Soz, left out the map